README.md

```diff
@@ -138,15 +138,48 @@ The expected payload structure is as follows:
 
 ## Streamlit Application
 
-Once the deployment has completed, run the following command to start a streamlit application.
+To deploy streamlit application on AKS, we first build a Docker image for streamlit application and push the image to ACR.
 
-> Note: you need to first obtain the model ingress IP as described above.
+> Note: Run the following command from the root of the project.
+
+Verify that you are in the root of the project.
+
+```bash
+pwd
+
+/home/username/MindGPT
+```
+
+We build and push the streamlit application to ACR. This image will be used by Kubernetes deployment.
 
 ```bash
-export SELDON_INGRESS=<ingress IP from matcha>
-python -m streamlit run app/app.py
+acr_registry_uri=$(matcha get container-registry registry-url --output json | sed -n 's/.*"registry-url": "\(.*\)".*/\1/p')
+acr_registry_name=$(matcha get container-registry registry-name --output json | sed -n 's/.*"registry-name": "\(.*\)".*/\1/p')
+```
+
+```bash
+az acr login --name $acr_registry_name
+docker build -t $acr_registry_uri/mindgpt:latest -f app/Dockerfile .
+docker push $acr_registry_uri/mindgpt:latest
 ```
 
+Line number 19 in [streamlit-deployment.yaml](./infrastructure/streamlit_k8s/streamlit-deployment.yaml#L19) should be updated to the name Docker image pushed to ACR, in this case it will be of format `<name-of-acr-registry>.azurecr.io/mindgpt`.
+
+Next, we apply the Kubernetes manifest to deploy the streamlit application on AKS.
+
+```bash
+cd infrastructure/streamlit_app_k8s
+kubectl apply -f .
+```
+
+Finally, we verify the streamlit application. The command below should provide an IP address for the streamlit application.
+
+```bash
+kubectl get service streamlit-service -o jsonpath='{.status.loadBalancer.ingress[0].ip}'
+```
+
+Interact with the streamlit application by visiting the address.
```

Comment:
If you visit that URL in browser, you should be able to interact with the deployed streamlit application.